[CLOUD-2595] Add AMQ_DATA_DIR as parameter to templates where it was m…

RoddieKieley opened the pull request:
…issing.

See [CLOUD-2595](https://issues.jboss.org/browse/CLOUD-2595)

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [X] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [X] Pull Request contains link to the JIRA issue
- [X] Pull Request contains description of the issue
- [X] Pull Request does not include fixes for issues other than the main ticket
- [X] Attached commits represent units of work and are properly formatted
- [X] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [X] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

RoddieKieley committed:
CLOUD-2595: Add AMQ_DATA_DIR as parameter to templates where it was missing.

- https://issues.jboss.org/browse/CLOUD-2595

Signed-off-by: Roddie Kieley <rkieley@redhat.com>

vbusch commented on templates/amq-broker-71-statefulset-clustered.yaml:
```diff
@@ -279,6 +279,10 @@ parameters:
+- description: The directory to use for data storage
+  displayName: AMQ Data Directory
+  name: AMQ_DATA_DIR
+  value: /opt/amq/data  
```
Also need to update
```
          volumeMounts:
          - name: ${APPLICATION_NAME}-amq-pvol
            mountPath: /opt/amq/data
```
to use $DATA_DIR

vbusch commented on templates/amq-broker-71-custom.yaml:
```diff
@@ -128,6 +128,8 @@ objects:
           - name: AMQ_ALLOW_ANONYMOUS
             value: ${AMQ_ALLOW_ANONYMOUS}
+          - name: AMQ_DATA_DIR
+            value: ${AMQ_DATA_DIR}
```
If we add the env, we will also need to add a parameter with a default.  However, I believe it best to not include it in templates that don't use statefulsets.  As long as it isn't listed as an env, the data dir parameter won't be included, and the default location will be used.

vbusch replied:
Creating a PR, to modify the above.

vbusch reviewed (changes requested):
For the following: amq-broker-71-basic.yaml, amq-broker-71-ssl.yaml, amq-broker-71-custom.yaml I believe we shouldn't add AMQ_DATA_DIR.  Without them mounting a volume, there are very few location. that will have the correct permissions.  Therefore, I think it is best that we leave it in the default location.
And without persistence, there is little reason to move the data dir.

vbusch committed:
[CLOUD-2595] Updated for DATA_DIR
- used DATA_DIR to mount in amq-broker-71-statefulset-clustered.yaml
- deleted amq-broker-71-statefulset.yaml.  It was an early version of amq-broker-71-persistence.yaml
- removed DATA_DIR from non-statefulset templates:
-- Without mounting a volume, there are very few locations for a user to set the datadir to.  Best to use the default one provided by artemis instance.

RoddieKieley committed:
Merge pull request #1 from vbusch/CLOUD-2595

[CLOUD-2595] Updated for AMQ_DATA_DIR

vbusch commented:
@RoddieKieley  looks good to me now.
